### ISSUE TITLE
[$60] Calculate `daysWorked` on WP create or RB start/end updates

### ISSUE THREAD

**Issue by maxceem** [topcoder-platform/taas-apis#330]
- In the first versions we calculated `daysWorked` when we created Resource Bookings or updated start/end date as was implemented in https://www.topcoder.com/challenges/00080061-9cca-49fd-90d6-0ca64dc776fd
- Later, we stopped re-calculating `daysWorked` and keep it `null` as per https://github.com/topcoder-platform/taas-apis/issues/200
- And now we would like to again calculate `daysWorked` when we created Resource Bookings or updated start/end date as per the logic described in the updated wiki document https://github.com/topcoder-platform/taas-apis/wiki/Work-Periods-(WP)-Automation-and-Constraints

The reason for this: is because logic with `null` is very hard to follow and creates a lot of difficult moments. So disadvantages of recalculating `daysWorked ` seems less strong then keeping it as `null`.

The new logic is describe in the updated wiki document https://github.com/topcoder-platform/taas-apis/wiki/Work-Periods-(WP)-Automation-and-Constraints.

**Comment by maxceem** [comment]
@eisbilir would you like to pick up this issue?

**Comment by maxceem** [comment]
Paid by https://www.topcoder.com/challenges/fcfa8b59-414f-4c23-b87f-29fa775d8f80

**Comment by sandhiyakavi** [comment]
@maxceem Verified daysWorked calculation and found 2 discrepancies below,
RB id: c202708a-7c58-423b-b595-b53ac3725d39 (06/02/2021-06/16/2021) has 3 WP's,
1. id: 5b9f3afd-20f9-450b-a39b-4d64043197f0, daysWorked=3
2. id: bd726bd4-fe8b-4271-86d1-8ff876b5a7be, daysWorked=5
3. id: 62c7c968-8974-4b5a-b2b0-c4e047f5889f, daysWorked=3

Set daysWorked=3 for middle WP
![image](https://user-images.githubusercontent.com/55895538/122884950-989f1c80-d379-11eb-9e22-526f4c822b12.png)

If change RB start date to later by weeks(new daysworked > daysworked set) -->06/08/2021 then No change in daysWorked(same 3 instead of 4 )

![image](https://user-images.githubusercontent.com/55895538/122885919-78239200-d37a-11eb-9870-b917cabd9e06.png)

Similar situation is happening if daysWorked set (for second WP from last) and then Enddate set earlier by weeks(new daysworked > daysworked set) -->no change in  daysWorked.

Other than the above two, below scenarios are working fine and as expected.
1. If daysWorked set for WP in the middle  --> no change in daysWorked.
2. If daysWorked set (for second WP) and then startdate set later by weeks(new daysworked < daysworked set) --> daysWorked updated according to RB start date.
3. If daysWorked set (for second WP from last) and then Enddate set earlier by weeks(new daysworked < daysworked set) --> daysWorked updated according to RB start date.

**Comment by maxceem** [comment]
@sandhiyakavi regarding these failed cases it looks like I misled you when I clarify this behavior for you in Slack.

The expected behavior is to only update `daysWorked` in such case if the current value is not possible:

![image](https://user-images.githubusercontent.com/146016/122893076-a7bca500-d34e-11eb-9c9d-d8aad584022f.png)

Sorry for the confusion.

**Comment by sandhiyakavi** [comment]
Verified on Prod Env with @nkumar-topcoder .
